judo-runtime-core-query/src/main/java/hu/blackbelt/judo/runtime/core/query/QueryFactory.java

```diff
@@ -346,7 +362,10 @@ private void createNavigations() {
                     dataQueries.put(attribute, subSelect);
                 });
 
-        getAsmElement(EReference.class)
+        Set<EReference> referencesToProcess = getAsmElement(EReference.class)
+                .filter(r -> asmUtils.isMappedTransferObjectType(r.getEReferenceType()) && r.isDerived() && !AsmUtils.isEntityType(r.getEContainingClass()) && isStaticReference(r)).collect(Collectors.toSet());
+
+        referencesToProcess.parallelStream()
                 .filter(r -> asmUtils.isMappedTransferObjectType(r.getEReferenceType()) && r.isDerived() && !AsmUtils.isEntityType(r.getEContainingClass()) && isStaticReference(r))
```

Comment:
Doesn't `referencesToProcess` already have all the items needed for this filter?